ckanext/validation/tests/test_form.py

```diff
@@ -93,288 +98,172 @@ def _post(app, url, data, resource_id='', upload=None):
 @pytest.mark.usefixtures("clean_db", "validation_setup")
 class TestResourceSchemaForm(object):
 
-    def setup(self):
-        self.app = helpers._get_test_app()
-
-    def test_resource_form_includes_json_fields(self, app):
+    def test_resource_form_includes_schema_fields(self, app):
+        """All schema related fields must be in the resource form"""
         dataset = Dataset()
 
         response = _get_resource_new_page_as_sysadmin(app, dataset['id'])
         form = _get_form(response)
 
         assert form.find("input", attrs={'name': 'schema'})
+        assert form.find("input", attrs={'name': 'schema_upload'})
         assert form.find("textarea", attrs={'name': 'schema_json'})
         assert form.find("input", attrs={'name': 'schema_url'})
 
-    def test_resource_form_create(self, app):
+    def test_resource_form_create_with_schema(self, app):
+        """Test we are able to create a resource with a schema"""
         dataset = Dataset()
 
-        value = {
-            'fields': [
-                {'name': 'code'},
-                {'name': 'department'}
-            ]
-        }
-        json_value = json.dumps(value)
-
-        params = {
+        data = {
             'name': 'test_resource_form_create',
             'package_id': dataset['id'],
-            'url': 'https://example.com/data.csv',
-            'schema': json_value,
+            'schema': json.dumps(SCHEMA),
         }
 
-        _post(app, NEW_RESOURCE_URL.format(dataset['id']), params)
+        _post(app, NEW_RESOURCE_URL.format(dataset['id']), data)
 
         dataset = call_action('package_show', id=dataset['id'])
 
-        assert dataset['resources'][0]['schema'] == value
+        assert dataset['resources'][0]['schema'] == SCHEMA
 
-    def test_resource_form_create_json(self, app):
+    def test_resource_form_create_schema_from_schema_json(self, app):
+        """Test we are able to create a resource with schema from a json"""
         dataset = Dataset()
 
-        value = {
-            'fields': [
-                {'name': 'code'},
-                {'name': 'department'}
-            ]
-        }
-        json_value = json.dumps(value)
-
-        params = {
+        data = {
```

Comment:
could you not have kept 'params' instead of changing to 'data' so it aligns on legacy code better.

Reply:
\+